modules/core/02-client/keeper/migrations.go

```diff
@@ -25,3 +26,13 @@ func NewMigrator(keeper Keeper) Migrator {
 func (m Migrator) Migrate1to2(ctx sdk.Context) error {
 	return v100.MigrateStore(ctx, m.keeper.storeKey, m.keeper.cdc)
 }
+
+// Migrate2to3 migrates from version 2 to 3.
+// This migration
+// - migrates solo machine client states from v2 to v3 protobuf definition
+// - prunes solo machine consensus states
+// - removes the localhost client
+// - asserts that existing tendermint clients are properly registered on the chain codec
+func (m Migrator) Migrate2to3(ctx sdk.Context) error {
```

Comment:
I have a question about the `Migrator` struct, do we need to this to be a struct (maybe implementing some interface?)

It looks like we only use 2 fields from the keeper and don't actually use any of the other keeper functionality. (cdc/storeKey)

I'm happy to stick with the pattern of adding functions, just curious if we need this struct!

Reply:
This is a good question. This was the approach documented by the [SDK](https://docs.cosmos.network/main/building-modules/upgrade)

> Since these migrations are functions that need access to a Keeper's store, use a wrapper around the keepers called Migrator as shown in this example:

I think it makes sense for the Keeper not to expose `Migrate2to3`, but I'm totally open to a different design